src/fromager/vendor_rust.py

```diff
@@ -86,28 +86,27 @@ def _cargo_config(project_dir: pathlib.Path) -> None:
         tomlkit.dump(cfg, f)
 
 
-def _should_vendor_rust(req: Requirement, project_dir: pathlib.Path) -> bool:
+def _detect_rust_build_backend(
+    req: Requirement, pyproject_toml: dict[str, typing.Any]
+) -> str | None:
```

Comment:
The docstring could use an update to explain the return value.

Reply:
I have updated the doc string and changed the return values to `enum | str`.